### PR TITLE
Optimize the cleanup of the map.

### PR DESCRIPTION
**Pull request by lctking**
**Optimize the cleanup of the map.
Check if the map is empty. If it's not empty, call the native cleanup method of the map; if it is empty, assign a value.**

Thanks for your contribution to [Apache Commons](https://commons.apache.org/)! Your help is appreciated!

Before you push a pull request, review this list:

- [x] Read the [contribution guidelines](CONTRIBUTING.md) for this project.
- [x] Run a successful build using the default [Maven](https://maven.apache.org/) goal with `mvn`; that's `mvn` on the command line by itself.
- [ ] Write unit tests that match behavioral changes, where the tests fail if the changes to the runtime are not applied. This may not always be possible but is a best-practice.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Each commit in the pull request should have a meaningful subject line and body. Note that commits might be squashed by a maintainer on merge.

### CODE REVIEW

**Comment by garydgregory** [comment]
@lctking
-1 You did not follow the instructions, even though you checked the box for "Run a successful build using the default [Maven](https://maven.apache.org/) goal with mvn; that's mvn on the command line by itself."

File: src/main/java/org/apache/commons/validator/Validator.java [hunk 1]
```diff
@@ -196,7 +196,11 @@ public Validator(final ValidatorResources resources, final String formName, fina
     public void clear() {
         formName = null;
         fieldName = null;
-        parameters = new HashMap<>();
+        if(parameters == null){
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/validator/Validator.java:199]: How can `parameters` ever be null?

**Comment by garydgregory** [comment]
@lctking 
Thank you for the PR but as noted, it cannot be accepted as is. Instead, I pushed new getters to git master and deprecated access to the instance variables directly. You are credited in changes.xml.

File: src/main/java/org/apache/commons/validator/Validator.java (hunk 1 above)
Comment (pzygielo) [reply src/main/java/org/apache/commons/validator/Validator.java:199]: - https://github.com/apache/commons-validator/commit/2d983f2f2a5e5523754ff88c611250204e2dbf66#r150814815

**Pull request closed without merging**